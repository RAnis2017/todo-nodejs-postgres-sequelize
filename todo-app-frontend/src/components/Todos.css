.addTodoList {
    display: flex;
    justify-content: center;
    align-items: center;
    column-gap: 1rem;
    flex-direction: row;
}

.addTodoList input {
    width: auto;
}

.tasksList {
    flex: 1 1 auto;
    display: flex;
    padding: 1rem 1rem;
    flex-direction: column;
    align-items: center;
    align-content: center;
}

.tasksList .form-check {
    width: fit-content;
}