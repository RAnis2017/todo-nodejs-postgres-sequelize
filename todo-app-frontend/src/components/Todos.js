import Context from '../store/Context';
import React from 'react';
import { Input, Button } from 'reactstrap'
import './Todos.css'
import TodoList from './TodoList';

const Todos = () => {
    return <Context.Consumer>
        {context => (
            <>
                <h3>Todo App</h3>

                <div className="addTodoList">
                    <Input />

                    <Button
                        color="primary"
                    >
                        New List
                    </Button>
                </div>
                {
                    <TodoList todos={context.todos} updateTodoStatus={context.updateTodoStatus} />
                }
            </>
        )}
    </Context.Consumer>
};

export default Todos;