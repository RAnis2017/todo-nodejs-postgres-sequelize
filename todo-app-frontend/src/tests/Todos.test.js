import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Todos from '../components/Todos'

describe('Todo Component', () => {

    beforeAll(() => {
        Enzyme.configure({ adapter: new Adapter() })
    })

    it('render correctly', () => {
        const component = shallow(<Todos />)
        expect(component).toMatchSnapshot()
    })

    describe('new todo list is made', () => {

        it('input for new todo list working', () => {
            const component = mount(<Todos />)
            component.find('.addTodoList > .inputText').simulate("change", {target: {value: 'New Test Todo List'}})
            expect(component.find('.addTodoList > .inputText').get(0).props.value).toEqual('New Test Todo List')
        })

        it('button click on input for new todo list working', () => {
            const component = mount(<Todos />)
            component.find('.addTodoList > .inputText').simulate("change", {target: {value: 'New Test Todo List'}})
            expect(component.find('.addTodoList > .inputText').get(0).props.value).toEqual('New Test Todo List')

            component.find('.addTodoList > .inputButton').simulate('click');

            expect(component.find('.addTodoList > .inputText').get(0).props.value).toEqual('')
        })
    })
})