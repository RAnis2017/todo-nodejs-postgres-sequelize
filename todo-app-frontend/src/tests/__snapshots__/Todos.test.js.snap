// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Component render correctly 1`] = `ShallowWrapper {}`;
